### PR TITLE
Playbook parsing: Separate disk read and parse from main function, add first tests

### PR DESCRIPTION
**Pull request by therealplato**
- Move functionality out of main into `ReadPlaybookFromDisk` and `ParsePlaybook`
- Rename `ValidateManifests` to `ManifestsPresent` (as it's just checking if the file exists on disk for now)
- Test `ReadPlaybookFromDisk` with a mock playbook yaml file
- Test `ParsePlaybook`
- Test `ManifestsPresent`

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -41,31 +41,41 @@ func (t Task) ValidateManifests() error {
...
 			return err
 		}
 	}
 	return nil
 }
 
+func ParsePlaybook(playbook []byte) (Playbook, error) {
+	var p Playbook
+	err := yaml.Unmarshal(playbook, &p)
+	return p, err
```
Comment (therealplato) [inline main.go]: is it correct to return `p` here instead of `&p`?

Comment (macat) [inline main.go]: yes, that'll copy the value instead of returning a pointer.

File: main.go [hunk 2]
```diff
@@ -41,31 +41,41 @@ func (t Task) ValidateManifests() error {
...
+}
+
+func ReadPlaybookFromDisk(fd string) ([]byte, error) {
+	file, err := ioutil.ReadFile(fd)
+	return file, err
```
Comment (bobbytables) [inline main.go]: probably could just do `return ioutil.ReadFile(fd)` instead of assignment then return.

File: main.go [hunk 3]
```diff
@@ -41,31 +41,41 @@ func (t Task) ValidateManifests() error {
...
+}
+
 func main() {
 	args := os.Args
-	yamlFile := args[1:][0]
+	yamlFileDescriptor := args[1:][0]
```
Comment (bobbytables) [inline main.go]: I think we should move to codegangsta for our CLI tool. It makes parsing arguments, help messages, etc, super easy to do.

**Comment by poopoothegorilla** [comment]
should we update these to match the other testing style in @macat code using testify?

**Comment by macat** [comment]
+1

**Comment by poopoothegorilla** [comment]
+1

**Pull request merged**